app/models/user.rb

```diff
@@ -1,3 +1,11 @@
 class User < ApplicationRecord
   validates_presence_of :email, uniqueness: { case_sensitive: false }, format: { with: Devise.email_regexp }
```

Comment:
hi. I'm the author of the `magic-link` gem. I think your issues are related to the User model. It doesn't look like you set it up using `rails g devise User` like the Devise setup instructions tell you to do. This class is missing a bunch of Devise configuration that is needed for it to work properly. Also, the errors you mentioned in the issue you posted are probably being caused by your `self.serialize_into_session` method which looks like custom code and is not something I can help you with. Here's an example of what a standard Devise User model looks like:

```
class User < ApplicationRecord
  # Include default devise modules. Others available are:
  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
  devise :database_authenticatable, :registerable,
         :recoverable, :rememberable, :validatable
end
```

Reply:
Well I was thinking that it may be connected to this, was not sure because I thought that I created an User model via `rails g devise User`. Will try your solution, thanks